packages/components/psammead-navigation/src/SwipeableNavigation/index.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useRef, useState } from 'react';
+import styled, { css } from 'styled-components';
+import { node, oneOf } from 'prop-types';
+import { GEL_GROUP_2_SCREEN_WIDTH_MAX } from '@bbc/gel-foundations/breakpoints';
+import useWindowWidth from '../../hooks/useWindowWidth';
+import useOverflowed from '../../hooks/useOverflowed';
+import useMediaQuery from '../../hooks/useMediaQuery';
+
+/* Convert C_POSTBOX to rgba as IE doesn't like 8 digit hex */
+const C_POSTBOX_TRANSPARENT = `rgba(184, 0, 0, 0)`;
+const C_POSTBOX_OPAQUE = `rgba(184, 0, 0, 1)`;
+
+const StyledSwipeableNav = styled.div`
+  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
+    white-space: nowrap;
+    overflow-x: scroll;
+    scroll-behavior: smooth;
+    -webkit-overflow-scrolling: touch;
+
+    /* Hide scrollbar */
+    scrollbar-width: none;
+    -ms-overflow-style: none;
+    &::-webkit-scrollbar {
+      display: none;
+    }
+
+    ${({ overflowed, scrollEnd }) =>
+      overflowed &&
+      !scrollEnd &&
+      css`
+        &:before {
+          content: ' ';
+          height: 100%;
+          width: 3rem;
+          position: absolute;
+          ${({ dir }) => css`
+            ${dir === 'ltr' ? 'right' : 'left'}: 0;
+          `}
+          bottom: 0;
+          z-index: 3;
+          overflow: hidden;
+          pointer-events: none; /* ignore clicks */
+          background: linear-gradient(
+            ${({ dir }) => (dir === 'ltr' ? 'to right' : 'to left')},
+            ${C_POSTBOX_TRANSPARENT} 0%,
+            ${C_POSTBOX_OPAQUE} 100%
+          );
+        }
+      `}
+  }
+`;
+
+const SwipeableNavigation = ({ children, dir }) => {
+  const isSwipeable = useMediaQuery('(max-width: 600px)');
+  const ariaHidden = isSwipeable && { 'aria-hidden': true };
```

Comment:
As far as I know, for now a non-js navigation is just a swipeable one with dropdown disabled. In that case should we have an option to set `aria-hidden` to false so that tabbing is not disabled? Or is this too much of an exception case that we will not consider?

Reply:
That's something to consider - on the same note, what happens at the moment during SSR? The useMediaQuery hook uses window.matchMedia - does this fall over during server rendering? Storybook just renders on the client doesn't it?

Reply:
We can't know what the width of the page is on server render, so `isSwipeable` will be undefined on the server and then set on the client - so as is in the example @OlgaLyubin mentioned, aria-hidden will not be set. This will also be the case for AMP - so it's something to think about for that too

Reply:
As discussed on Slack, if the `window` is not defined the `useMediaQuery` will return null so `isSwipeable` will be null too, and we won't be applying `aria-hidden: true` and `tabIndex: -1`, therefor it will remain accessible when non-js.

Looking now at AMP.

Reply:
Should this comment be put in the accessibility section of the readme or a comment in the code? It's not an obvious result of the code so might be worth capturing the result.